floater/generators.py

```diff
@@ -76,6 +81,16 @@ def __init__(self, xlim=None, ylim=None, dx=1., dy=1., model_grid=None, load_pat
             self.Ny = int(self.Ly / self.dy)
             self.x = self.xlim[0] + self.dx * np.arange(self.Nx) + self.dx/2
             self.y = self.ylim[0] + self.dy * np.arange(self.Ny) + self.dy/2
+            try:
+                zvect[0]
+            except (NameError,TypeError,AttributeError,KeyError) as e:
+                self.z=np.asarray(-0.5)
+                self.Nz=1
+                self.zvect=[-0.5]
+            else:
+                self.zvect = zvect
+                self.z = zvect
```

Comment:
What is the difference between `self.z` and `self.zvect`? As far as I can tell, they are always the same. Do we really need both?